libs/sdk/skyux-eslint/src/rules/template/prefer-disabled-attr.spec.ts

```diff
@@ -0,0 +1,239 @@
+import { convertAnnotatedSourceToFailureCase } from '@angular-eslint/test-utils';
+
+import { createTemplateRuleTester } from '../testing/create-template-rule-tester';
+
+import { RULE_NAME, messageId, rule } from './prefer-disabled-attr';
+
+const ruleTester = createTemplateRuleTester();
+
+ruleTester.run(RULE_NAME, rule, {
+  valid: [
+    // Elements that don't support disabled attribute should be ignored
+    `<div class="sky-btn-disabled"></div>`,
+    `<span class="sky-btn-disabled"></span>`,
+    `<p class="sky-btn-disabled"></p>`,
+    `<div [class.sky-btn-disabled]="isDisabled">Div with disabled class</div>`,
+    `<span [class]="'sky-btn-disabled'">Span with class binding</span>`,
+    `<div [ngClass]="{'sky-btn-disabled': isDisabled}">Div with ngClass</div>`,
+
+    // Elements without sky-btn-disabled class are valid
+    `<button class="other-class"></button>`,
+    `<input class="btn btn-primary" />`,
+
+    // Elements with disabled attribute but NO sky-btn-disabled class are valid
+    `<button disabled></button>`,
+    `<input [disabled]="isDisabled" />`,
+    `<textarea [attr.disabled]="condition"></textarea>`,
+    `<select disabled></select>`,
+
+    // Empty class attributes
+    `<button class=""></button>`,
+    `<input class="" />`,
+  ],
+  invalid: [
+    // sky-btn-disabled class should NEVER be used on elements that support disabled
+    convertAnnotatedSourceToFailureCase({
+      description:
+        'should fail when button has sky-btn-disabled class (no disabled attribute)',
+      annotatedSource: `
+      <button class="sky-btn-disabled"></button>
+              ~~~~~~~~~~~~~~~~~~~~~~~~
+      `,
+      annotatedOutput: `
+      <button disabled></button>
+              ~
+      `,
+      messageId,
+      data: {
+        element: 'button',
+      },
+    }),
+    convertAnnotatedSourceToFailureCase({
+      description:
+        'should fail when button has sky-btn-disabled class WITH disabled attribute',
+      annotatedSource: `
+      <button class="sky-btn-disabled" disabled></button>
+              ~~~~~~~~~~~~~~~~~~~~~~~~
+      `,
+      annotatedOutput: `
+      <button disabled></button>
+              ~
+      `,
+      messageId,
+      data: {
+        element: 'button',
+      },
+    }),
+    convertAnnotatedSourceToFailureCase({
+      description:
+        'should fail when button has sky-btn-disabled class WITH property binding',
+      annotatedSource: `
+      <button class="sky-btn-disabled" [disabled]="isDisabled"></button>
```

Comment:
Nitpick - sorry if I missed it - can you add a test that ensures that when `sky-btn-disabled` is removed that another class such as `sky-btn` is retained?

Reply:
Look for the test named, "should fail with multiple classes including sky-btn-disabled".